### ISSUE TITLE
Two Way Sync?

### ISSUE THREAD

**Issue by edent** [owncloud/android#1437]
### Expected behaviour
When I hit the "Synchronize" menu item, files stored locally on my device should be uploaded to the server.

Ideally, this should happen automatically.

### Actual behaviour
Files are downloaded from the server, but local changes are not uploaded.

### Steps to reproduce
1. Open Android app.
2. Long press on a folder.
3. Select "Synchronize"
4. The folder is stored in the local filesystem `/storage/emulated/0/owncloud/user@server/foldername/`
5. Edit one of the downloaded files - or save a new file in the folder.
6. Changes and new files are not automatically syncronised.
7. Repeat step 3.
8. Changes and new files are not automatically syncronised.

I've gone into Android's system settings - accounts - owncloud - and enabled "Sync Files".  The problem still persists.

I suppose I want the same functionality as the Linux Desktop client - the ability to selectively sync folders.  Is that possible?

If not, can this menu item be renamed "Download" rather than "Synchronize"?

### Environment data
Android version: 6.0.1

Device model: Nexus 6

Stock or customized system:  CM13

ownCloud app version: 1.9

ownCloud server version: 8.2

#### Web server error log
N/A

#### ownCloud log (data/owncloud.log)
N/A

**Comment by abishai** [comment]
Must have feature. Without it the app is useless for me as I'm looking how to sync call recordings to owncloud and the app has no ability to sync user choosen folder in 2 way.

**Comment by masensio** [comment]
Hi @edent,

thanks for your feedback.

The menu option Synchonize launch a manual sync process. This option doesn't  work as Desktop Sync option. When you select this option the folder is syncronized only once, there is no folder indication that says it should be automatically synchronized.
The option for automatic synchronization is to set the folder as Favourite. This feature is in our roadmap, and it will be available in a near future

One doubt about your issue, are you saving a new file in the folder using outside owncloud app? If you access to the folder using another app and add a file directly, ownCloud App doesn't know that this file exists and it won't be included in the sync process.

**Comment by abishai** [comment]
Why not? I thought this is the core of 2-way sync - ability to sync folders with owncloud. I'm using syncthing for that now and I'd like idea to use only 1 application for all cases.
Andoid app allows to sync videos and capera shots, why the support can't be extended to user choosen folders?

**Comment by grantdobbe** [comment]
My experience is different -- files don't automatically sync from ownCloud, but do automatically sync *to* ownCloud if they're changed.

Example: I use todo.txt to manage my personal to-do and grocery list. I use ownCloud to synchronize it between my desktop PC, my laptop, and my Android phone. 

When I update todo.txt on my laptop, it is automatically synced to ownCloud and then pulled down to my desktop by the ownCloud client running on that machine. The same is true when I make edits to todo.txt on my desktop; when my laptop is on and connected to a network, the ownCloud client syncs the changes.

If I want to get the most recent version of todo.txt on my phone (for example, when I go to the grocery store), 90% of the time I have to manually force synchronize it in the ownCloud app, even though the file is "favorited". However, when I make changes to the file on my phone, the ownCloud app automatically syncs the changes to my ownCloud server. 

By comparison, when I used the Dropbox app for this functionality, synchronization was seamless -- I'd change the file on a computer, and (assuming I was connected to a network), the file would appear on my phone within seconds.

What I'd like to see is an option in the ownCloud app to change the interval at which the server is polled; that way, even if the app isn't constantly connected to the server, I still have a reasonable assurance that a file will be updated on a regular basis.

**Comment by Paviluf** [comment]
I don't have the exact same behavour as @grantdobbe :

I also use a todo.txt. I use ownCloud to synchronize it between my desktop PC, my laptop, and my Android phone.

When I update todo.txt on my laptop, it is automatically synced to ownCloud and then pulled down to my desktop by the ownCloud client running on that machine. The same is true when I make edits to todo.txt on my desktop; when my laptop is on and connected to a network, the ownCloud client syncs the changes.

If I want to get the most recent version of todo.txt on my phone (for example, when I go to the grocery store), **[100% of the time I have to manually force synchronize it](https://github.com/owncloud/android/issues/333)** in the ownCloud app. When I make changes to the file on my phone, **the ownCloud app don't automatically sync it and I need to manually sync the file for the changes to be updated to my ownCloud server.**

**Comment by keitalbame** [comment]
Also had the issue that any file that was changed on the phone, wasn't synced to the server or to desktop client.
Only after I marked as favourite, it started to upload the changes to the server.

Although some file extensions (kbdx - keepassx), when opened from outside owncloud app on the phone, does not upload automatically. Switching to owncloud app will trigger the upload (no need to refresh manually).
.txt files when opened from outside owncloud app, are updated automatically (no need to switch wo owncloud app).

IMO the `Favourite` mention should be renamed to `2 way sync` or something more explicit.

Archlinux, Nexus 6 with updated vanilla ROM (M)

**Comment by strk** [comment]
@masensio would setting the root folder as "Favorite" be possible ? It would then give the same experience as the desktop client, right ?

**Comment by Paviluf** [comment]
When I make changes to a file on my phone, can the ownCloud app automatically sync it please.
It's not practical at all :(

**Comment by helmo** [comment]
The main issue to follow at the moment is #1677 

I'd say this is a duplicate of #1136 and relates to #1663

**Comment by davivel** [comment]
Thanks, @helmo , you summarized it really nice. 

@edent, with the current version of the app, any time you repeat the step 3, the files that were changed in that folder should be uploaded; but the new files in the folder will not be uploaded. That's something we didn't consider yet because file browsers in mobile systems are not a built-in feature as they are in desktop systems, and the use case of adding files to the local ownCloud folder via any app that is not the OC app, with no interaction with the OC app, is not obvious at all for not-tech-savvy users. We could consider it later, but there are other details about sync we need to solve first.

What we plan to do with #1136 is allow the user to select what folders should receive the current 'Synchronize' operation in the menu **automatically** instead of with a direct interaction. 

Besides, we plan also in #1683 that local changes in downloaded files are **automatically** detected and uploaded even for files not marked as "available offline" (currently, "favorites"). The difference between "available offline" and not-"available offline" files will be kept for the update of remote changes. #333 deals also with this.

Let's observe all these changes in the aforementioned issues. Take #1677 as the entry point for the group.

And, if you are really interested in these improvements, take the chance of voting for them in https://github.com/owncloud/core/issues/24684#issuecomment-220038870 . Let us see your hearts there!!

**Comment by moolight-seashell** [comment]
2016 and still not possible to sync file ? 

and make aviable offline just store file as internal app data so unaviable on filesystem ? 

have i missed something ? when i sync a file, i have an already sync message and this file does not exist on all the filesystem .

**Comment by moolight-seashell** [comment]
> .@moolight-seashell sync a file means:
> 
>     * File is downloaded with the latest version
> 
>     * If the file was already downloaded, new version is downloaded (if exists)
> 
>     * If there are local changes, these changes are pushed to the server
> 
>     * If the file is available offline, everything is done automatically
> 
>     * Synced files are always stored in the file system and available via SAF

where on filesystem, can i save it on my sdcard ?
is it only available from saf? no because 99.99% of apps and use case ar not using it, ( i try to sync obsidian note files so i need a real filesystem acess )

**Comment by jesmrec** [comment]
> where on filesystem, can i save it on my sdcard ?

you can do it via SAF, that's the recommended way to do it. In the app `Settings` we recommend to use an specific app to browse through oC (or any app with SAF support), in which you will be allowed to move your files from oC (or any app with SAF support) to your SD card or any other location in your device.

**Comment by moolight-seashell** [comment]
maybe i can copy them but i need to do this manually and it's not what i want . Saf It's the recommended way ? but Recommended doesn't mean only way. and how about all the users that d'ont want or cannot use saf . 

even when you think that this kind of cloud is intended for moslty for advanced users.
